envoy/ssl/connection.h

```diff
@@ -95,6 +96,12 @@ class ConnectionInfo {
    **/
   virtual const std::string& subjectPeerCertificate() const PURE;
 
+  /**
+   * @return the well-known attribute values parsed from subject field of the peer certificate.
+   *         Returns absl::nullopt if there is no peer certificate.
+   **/
+  virtual const ParsedX509NamePtr& parsedSubjectPeerCertificate() const PURE;
```

Comment:
Don't return a reference to a unique_ptr; that's a weird interface. Switch back to the OptRef you had.

Reply:
I actually like the reference to unique_ptr than the OptRef interface. 
1. the usage is no much difference from caller, just use as a pointer for both cases. 
2. using the reference to unique_ptr actually make code cleaner. 
3. since we named the unique_ptr to ParsedX509NamePtr now all the return type of interface in connectionInfo is const XXX& which is more consistent and less confusing
4. it has less chance to let caller hold an invalid pointer like the OptRef because a const unique_ptr is not allowed to copy/move in general.

Reply:
It's not the correct way to treat unique_ptr in c++. Please fix.